docs/features/nullable-metadata.md

```diff
@@ -41,14 +41,13 @@ represents nullability: 0 for oblivious, 1 for not annotated, and 2 for annotate
 The `byte[]` is constructed as follows:
 - Reference type: the nullability (0, 1, or 2), followed by the representation of the type arguments in order including containing types
 - Nullable value type: the representation of the type argument only
-- Value type: the representation of the type arguments in order including containing types
+- Non-generic value type: skipped
+- Generic value type: 0, followed by the representation of the type arguments in order including containing types
```

Comment:
> 0,  [](start = 21, length = 4)

just curious: why do we want to encode an annotation for a value type? Is this due to a concern for cycles when loading metadata?

Reply:
Yes, the reason that generic value types (and type parameters constrained to `struct`) include a `0` byte is to avoid cycles when loading types from metadata - in particular, when loading type parameters with constraint types. Simple (non-generic) value types are less of a concern.

I'll add a note to the document.

---
In reply to: [294943765](https://github.com/dotnet/roslyn/pull/36519#discussion_r294943765) [](ancestors = 294943765)